jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/StepsRunner.java

```diff
@@ -379,9 +384,9 @@ private void buildImages() {
                                     realizeFutures(Verify.verifyNotNull(entry.getValue())),
                                     realizeFutures(Verify.verifyNotNull(results.applicationLayers)))
                                 .call());
-                builtImages.add(builtImage);
+                builtImagesAndLayers.put(builtImage, entry.getValue());
```

Comment:
Since we won't use `builtImagesAndLayers` in `pushBaseImageLayers()` (see my other comment), perhaps we don't need to change `buildImages()` code?